pkg/promtail/server/server.go

```diff
@@ -169,9 +177,11 @@ func (s *Server) targets(rw http.ResponseWriter, req *http.Request) {
 
 // ready serves the ready endpoint
 func (s *Server) ready(rw http.ResponseWriter, _ *http.Request) {
-	if !s.tms.Ready() {
-		http.Error(rw, readinessProbeFailure, http.StatusInternalServerError)
-		return
+	if s.healthCheckTarget {
+		if !s.tms.Ready() {
```

Comment:
Sorry to snoop around here. You can just do `if s.healthCheckTarget && !s.tms.Ready()`. You don't need two `if` statements.